### DISCUSSION TITLE
4 Port 4K Video Capture Card USB 3.0 HDMI-compatible Grabber

### DISCUSSION THREAD

**Discussion by MattWestb** [awawa-dev/HyperHDR#964] [General]
Was baying one of this and its very interesting !!
![HDMI1x4A](https://github.com/user-attachments/assets/53a2052b-96a5-425f-b207-c84e04c33ec6)
I only using it in full HD and capture the sum signal that i going  to the AVR and then to the TV.
What is not working it ARC so cant putting it between the AVR and the TV thats little sad also CEC looks not being connected at all to the HDMI in/outputs so cant scanning active devices from HA.
From Armbian log:
```
usb 2-1: new high-speed USB device number 3 using xhci-hcd
usb 2-1: New USB device found, idVendor=345f, idProduct=2131, bcdDevice=21.00
usb 2-1: New USB device strings: Mfr=1, Product=5, SerialNumber=7
usb 2-1: Product: C7 USB3.0 Video
usb 2-1: Manufacturer: MACROSILICON
usb 2-1: SerialNumber: 20210621
usb 2-1: Found UVC 1.00 device C7 USB3.0 Video (345f:2131)
hid-generic 0003:345F:2131.0002: hiddev96,hidraw0: USB HID v1.10 Device [MACROSILICON C7 USB3.0 Video] on usb-xhci-hcd.0.auto-1/input4
```
I think its one MS2131 chip plus one switch before it.
Normally it working OK with USB and HDMI conected but some combinations its need extra 5V thru USB-C conector for not getting trubble with AV (i think some HDMI device is having not  good 5V supplay and forcing the other down).
Its looks  being one of the big problems with this USB-devices falling (have 3 RK3318 boxes and one working very good an all other have USB problems all the time).

Its very nice working on one TV-Box (Ambrian or Android with IP-TV) and at the same time have the TV news running in one window on the same screen !!

Some more have experience of this devices then pleas posting it here so other users can getting it working well with HyperHDR.

**Comment by awawa-dev** [comment]
Hi
> I think its one MS2131 chip plus one switch before it.

yes, that's what it looks like to me too. I also have a grabber with ms2131 but a simply one. Just curious how they got along with the internal splitter from the ms2131 which also modifies the EDID to suit its needs? And the most important question is does HDR passthrou work? And what's the price?

> Its looks being one of the big problems with this USB-devices falling (have 3 RK3318 boxes and one working very good an all other have USB problems all the time).

Oh yes, if you look through my blog you will find a whole list of problems TVBoxes have with USB3.0 and wifi and rk3318 are not particularly stable. That's why we like Raspberry Pi, even though they have been having kernel problems for some time now, I have the impression that the community is simply not as focused on their products anymore and fewer people are testing them and reporting feedback. Fortunately, N100 SoC board appeared and if I wanted to create a media center today I would choose e.g. Radxa x4 although the seller is playing around with the availability and price of this product and the BIOS is also locked (but RPi also uses locked blobs) which can lead to very big problems if he abuses our trust.

**Reply by MattWestb** [reply]
As you was very nice and doing the upgrade of **HyperSerialWLED** i must using the hammer little and you was right !!
![20241027_180633](https://github.com/user-attachments/assets/e5125059-73e4-4dec-9754-d11b1c32462e)
I was having problem with sound was dropping then connecting it to the TV and then have the Yamaha AVR fed with ARC but its buggy Samsung TV and old AVR (looks working OK now after hard repower AVR and TV).
![20241027_182332](https://github.com/user-attachments/assets/31ad7243-48b9-4938-8db2-a51542a185f0)
(_Flower lights is still burning and making all magenta_)
UCP Box IP-TV Box (Android 11 RK3318 with bad USB). Chrome cast and one Intel Atom (only 4G ram but 4 cores) running Lubuntu (All grabber works OK on it but i dont like have them on the server).
I was getting it under 40€ with shipping https://www.aliexpress.com/item/1005006354721356.html?spm=a2g0o.order_list.order_list_main.21.21ef1802pnnKLb
I was looking for one Amlogic S905X3 with 4G ram and also Gb Lan (using Eth all over then possible or G5 WiFi for not blocking Zigbee and Thread network on the 4.2 Ghz) but they is not easy to getting and not sure if Arbian running well and getting updated on them. So for the moment its looks like one E1000 with 8G ram and Gb Lan but its around 110€ shipped but then i can moving all from the Atome and using it as HyperHDR server.

I have changing the hardware settings then it was very wrong tuned (black very down under and contrast was very week like mist all over) but i dont knowing if all is right done and the captured video is very green.
Dont have any LUT for the card and was trying calibrating it with web-browser from the Android TV Box but its not possible getting the test in full screen and then clicking start the test.

Do you like /need more information only say and i trying getting it for you and great thanks for all work you have done !!!

**Reply by awawa-dev** [reply]
Now you can easily calibrate grabber with new methods, no need to use browser and Windows, just use test mp4 video files: https://github.com/awawa-dev/HyperHDR/pull/896 Latest installers are here: https://github.com/awawa-dev/HyperHDR/actions/runs/11533326015
If the grabber pass-through HDR I would be also interested in dumping firmware to test it with my ms2131 :-)

**Comment by MattWestb** [comment]
One update.
Was having very low contrast and grayish picture in combination with audio muting little now and then.
I have the cable-box connected with one cheep splitter one output to the TV and the other to the switch / grabber. 
Was pocking around and was finding that the Samsung TV was having the HDMI port input mode sett to "PC"  and the good working one to STB-Box. Was change the input mode and the picture quality is now OK as direct from the cable-box.

Then i was finding that the direct connected picture is 1920x1080 @ **50Hz** and thru the switch is 1920X1080 @ **60Hz**. That i think the AVR is getting problem with the times stamp the try have the audio and video i sync and doing clocking for 60Hz and the data is samples for 50Hz.
I dont knowing if the "re-scaling" is done of the grabbing chip or the mux / switch but somthing is doing it very wrong.
Was looking on the PCB and its one normal flash for the grabber ICs firmware plus one MCU that is connecting the buttons, switches and LEDs and then the HDMI switch / mux and its having internal firmware for its functionality like re-scaling for multi source view so can doing bad things to the signal.
More brainstorming is welcome for getting deeper in the problem if it possible to fixing it.
